pallets/payment/src/weights.rs

```diff
@@ -42,89 +45,89 @@ pub trait WeightInfo {
 	fn accept_and_pay() -> Weight;
 }
 
-/// Weight functions for `virto_payment`.
+/// Weights for virto_payment using the Substrate node and recommended hardware.
 pub struct SubstrateWeight<T>(PhantomData<T>);
 impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
-// Storage: Payment Payment (r:1 w:1)
+	// Storage: Payment Payment (r:1 w:1)
 	// Storage: Sudo Key (r:1 w:0)
 	// Storage: Assets Accounts (r:2 w:2)
 	// Storage: System Account (r:1 w:1)
 	fn pay() -> Weight {
-		(82_000_000 as Weight)
+		(54_000_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(5 as Weight))
 			.saturating_add(T::DbWeight::get().writes(4 as Weight))
 	}
 	// Storage: Payment Payment (r:1 w:1)
 	// Storage: Sudo Key (r:1 w:0)
 	// Storage: Assets Accounts (r:2 w:2)
 	// Storage: System Account (r:1 w:1)
-	fn pay_with_remark() -> Weight {
-		(76_000_000 as Weight)
+	fn pay_with_remark(_x: u32, ) -> Weight {
+		(54_397_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(5 as Weight))
 			.saturating_add(T::DbWeight::get().writes(4 as Weight))
```

Comment:
Thinking about it, the remark length, unless it's probably huge, makes sense that it doesn't make much difference in the benchmark since we don't do anything with the data other than store it, the db reads/writes stay the same. It will probably make a difference in the `FeeHandler` since users can do anything there, a custom FeeHandler can potentially run an expensive operation that parses the remark and operates on its data, so I guess we need to take make sure the computation in the handler is taken into account.

Reply:
Thats already there, the above line `// Storage: Sudo Key (r:1 w:0)` is connected to our fee handler [implementation](https://github.com/virto-network/virto-node/blob/5dd558572b446552b13e8dd91ca3dbd2fa83bb6d/runtime/src/lib.rs#L616). I'm afraid we cannot make it generic, ie. if the pallet is adopted by another project, they will have to re-run benchmarks according to their implementation.

Reply:
I think it's safe to expect the implementer will have their own custom benchmark? https://substrate.stackexchange.com/a/344/67